django/university/controllers/SigarraController.py

```diff
@@ -0,0 +1,114 @@
+import requests
+import json
+
+from datetime import date
+from tts_be.settings import CONFIG
+
+class SigarraResponse:
+    def __init__(self, data, status_code):
+        self.data = data
+        self.status_code = status_code
+    
+class SigarraController:
+    def __init__(self):
+        self.username = CONFIG["SIGARRA_USERNAME"]
+        self.password = CONFIG["SIGARRA_PASSWORD"]
+        self.cookies = None
+
+        self.login()
+    
+    def get_student_photo_url(self, nmec) -> str:
+        return f"https://sigarra.up.pt/feup/pt/fotografias_service.foto?pct_cod={nmec}"
+
+    def semester_weeks(self):
+        currdate = date.today()
+        year = str(currdate.year)
+        first_semester = currdate.month >= 9 and currdate.month <= 12
+        if first_semester: 
+            semana_ini = "1001"
+            semana_fim = "1201"
+        else:
+            semana_ini = "0101"
+            semana_fim = "0601"
```

Comment:
Start and end of semesters in order to use  the sigarra api.

Do you know if UNI does it differently?

Reply:
As discussed, use `period` query param.

UNI get them all, and just show the current week

Reply:
@tomaspalma Have you check this?

Reply:
Yes, I've been seeing the changes needed to switch to the new api and I it appears it is harder than expected, in term of what is needs to be done.

I saw how it is used in `uni` and it appears they first go to the user profile page in sigarra and extract the schedule url inside of the html and then make a request for it. I saw that in this [PR](https://github.com/NIAEFEUP/uni/pull/1281/files#diff-a20696808981d37998213e16bce388998bcd56d19079cb2b1bad934a2781a683) in [packages/uni_app/lib/controller/fetchers/schedule_fetcher/schedule_fetcher_new_api.dart](https://github.com/NIAEFEUP/uni/pull/1281/files#diff-e9a7ed5dac29edbe72ff8ebdfd2f8a6ca66b768b97bcf31a6ff8a58b371126a3).

As I understood this is because the correct url to fetch the schedule is like this, for example: https://sigarra.up.pt/calendarios-api/api/v1/events/feup/student/202108880/?academic_year=2024&pv_turma_ocorr=FEUP_540676_214441&pv_turma_ocorr=FEUP_540677_214444&pv_turma_ocorr=FEUP_540678_214440&pv_turma_ocorr=FEUP_540679_214443&pv_turma_ocorr=FEUP_540680_214440&period=1&period=2&period=3&period=4&period=5&period=6&period=7&period=8&lang=pt

I tried removing the `pv_turma_ocorr` parameters and it looks like it does not work without it.

From what I understood, switching to this API will involve making a request to the user profile and extract the correct url from the HTML. We can then store this url in a cookie in the user's browser so we do not have to always fetch from the url, but it maybe more complex than the current api.

I don't know if you have more insight related to this api since you are involved in the development of uni

Reply:
Oh, I see. Right, let's move this to a separated issue